rllib/algorithms/algorithm_config.py

```diff
@@ -3092,6 +3117,44 @@ def items(self):
         """Shim method to help pretend we are a dict."""
         return self.to_dict().items()
 
+    @staticmethod
+    def _serialize_dict(config):
+        # Serialize classes to classpaths:
+        config["callbacks"] = serialize_type(config["callbacks"])
+        config["sample_collector"] = serialize_type(config["sample_collector"])
+        if isinstance(config["env"], type):
+            config["env"] = serialize_type(config["env"])
+        if "replay_buffer_config" in config and (
+            isinstance(config["replay_buffer_config"].get("type"), type)
+        ):
+            config["replay_buffer_config"]["type"] = serialize_type(
+                config["replay_buffer_config"]["type"]
+            )
+        if isinstance(config["exploration_config"].get("type"), type):
+            config["exploration_config"]["type"] = serialize_type(
+                config["exploration_config"]["type"]
+            )
+        if isinstance(config["model"].get("custom_model"), type):
+            config["model"]["custom_model"] = serialize_type(
+                config["model"]["custom_model"]
+            )
+
+        # Serialize dataclasses.
+        if isinstance(config.get("_learner_hps"), LearnerHPs):
+            config["_learner_hps"] = dataclasses.asdict(config["_learner_hps"])
```

Comment:
learner_hps has its own to_dict method which we should opt to use here instead.

Reply:
Perfect, changed.

Reply:
ah it does not have it i was mistaken, apologies. but ill think about adding one in my pr for checkpointing learners.